src/main/java/seedu/socket/model/person/UniquePersonList.java

```diff
@@ -98,6 +99,16 @@ public void remove(Person toRemove) {
         }
     }
 
+    /**
+     * Removes all persons by the given {@code predicate}.
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    public boolean removeAll(Predicate<Person> predicate) {
+        requireNonNull(predicate);
+        boolean getRemoved = internalList.removeIf(predicate);
+        return getRemoved;
```

Comment:
Minor nitpick from me, but maybe would be better to name it isRemoved to better reflect boolean field, but if you think it is ok to keep the current name, can just ignore this comment.

Reply:
Oh i see, thanks for the suggestion, I would change it later